src/utilsAuth.js

```diff
@@ -0,0 +1,39 @@
+// Functions for checking user's auth mapping
```

Comment:
I noticed some code here doesn't have a property check / out-of-index check. It would be safe to handle those edge cases so that the website won't crash for null/undefined data. Thanks!

Reply:
Do you mean the string manipulation in `resourcePathFromProjectID`? If so: 
```
> split = ''.split('-')
[ '' ]
> split
[ '' ]
> program = split[0]
''
> project = split.slice(1).join('-')
''
> resourcePath = ["/programs", program, "projects", project].join('/')
'/programs//projects/'
>
```

Reply:
Or if you mean the `userAuthMapping`s, these should always be defined because of the `useArboristUI` checks, but if you want, I can add extra checks anyway?

Reply:
Oh yeah maybe just things like `x['method'] === 'file_upload' && x['service'] === 'fence'` makes me nervous but I should read code more carefully. it's all up to you~ :p

Reply:
Ah, that particular fn is passed to `Array.some`, so either the array is empty and the fn isn't called at all, or the array is a list of actions from Arborist (so those fields are guaranteed by the API)--if the Arborist API changes, we should just be updating this code 😄  But I added checks for `userAuthMapping`!